src/pages/agent/AgentTemplateDeliveredOrderContext.tsx

```diff
@@ -0,0 +1,96 @@
+import { createContext, PropsWithChildren, useContext, useState } from "react";
+import { disableDeliveredOrderTracking, enableDeliveredOrderTracking } from "../../services/agent.service";
+import { toast } from "@vtex/shoreline";
+
+export const AgentTemplateDeliveredOrderContext = createContext<{
+  appKey: string;
+  setAppKey: (appKey: string) => void;
+  appToken: string;
+  setAppToken: (appToken: string) => void;
+  isEnabled: boolean;
+  enable: (agentUuid: string) => Promise<void>;
+  isEnabling: boolean;
+  disable: (agentUuid: string) => Promise<void>;
+  isDisabling: boolean;
+  setIsEnabled: (isEnabled: boolean) => void;
```

Comment:
Something like an enum here would not be better to handle this state? Something like: Status.ENABLED, Status.ENABLING, Status.DISABLED
And only one method would be required to change this.